### PR TITLE
Bug 1827073: Support mirroring catalogs using default-tagged source images.

### PR DESCRIPTION
**Pull request by benluddy**
Catalogs may reference image pull specs without a tag or digest, with
the expectation that the tag "latest" will be assumed.

### CODE REVIEW

File: pkg/cli/admin/catalog/mirrorer_test.go [hunk 1]
```diff
@@ -169,11 +169,11 @@ func TestMirror(t *testing.T) {
 					WithTag:    "quay.io/org/coreos-prometheus-operator:76771fef",
 					WithDigest: "quay.io/org/coreos-prometheus-operator@sha256:5037b4e90dbb03ebdefaa547ddf6a1f748c8eeebeedf6b9d9f0913ad662b5731",
 				},
-				"quay.io/test/etcd.0.9.2": {
-					WithTag: "quay.io/org/test-etcd.0.9.2:f0e557b2",
```
Comment (awgreene) [inline pkg/cli/admin/catalog/mirrorer_test.go:173]: Was there a reason you opted to update existing tests rather than create new tests?

Comment (benluddy) [reply pkg/cli/admin/catalog/mirrorer_test.go:173]: This changes the existing behavior -- mappings from name-only sources get an explicit "latest" tag -- so the affected test cases needed to have their expectations updated.

**Pull request merged**